build-support/docker/centos6/Dockerfile

```diff
@@ -19,24 +19,35 @@ RUN yum install -y \
         openssl-devel \
         readline-devel \
         sqlite-devel \
+        unzip \
         zlib-devel
 
 ARG PYTHON_27_VERSION=2.7.13
 ARG PYTHON_36_VERSION=3.6.8
-# NB: PYENV_ROOT must be set for `pyenv install` to be available. This failure mode is not mentioned
-# in https://github.com/pyenv/pyenv#basic-github-checkout.
+
 ENV PYENV_ROOT /pyenv-docker-build
 ENV PYENV_BIN "${PYENV_ROOT}/bin/pyenv"
 RUN git clone https://github.com/pyenv/pyenv ${PYENV_ROOT}
 
-# Install Python 2.7 and 3.6.
 # NB: We intentionally do not use `--enable-shared`, as it results in our shipped wheels for the PEX release using
 # `libpython.X.Y.so.1`, which means that the PEX will not work for any consumer interpreters that were statically
 # built, i.e. compiled without `--enable-shared`. See https://github.com/pantsbuild/pants/pull/7794.
 RUN /usr/bin/scl enable devtoolset-7 -- ${PYENV_BIN} install ${PYTHON_27_VERSION}
 RUN /usr/bin/scl enable devtoolset-7 -- ${PYENV_BIN} install ${PYTHON_36_VERSION}
-RUN ${PYENV_BIN} global ${PYTHON_27_VERSION} ${PYTHON_36_VERSION}
-ENV PATH "${PYENV_ROOT}/shims:${PATH}"
+
+ENV PATH "${PYENV_ROOT}/versions/${PYTHON_27_VERSION}/bin:${PATH}"
+ENV PATH "${PYENV_ROOT}/versions/${PYTHON_36_VERSION}/bin:${PATH}"
+
+# We install the AWS CLI for use in CI to cache assets like pants.pex.
+ENV AWS_CLI_ROOT /aws-docker-build
+ENV AWS_CLI_BIN "${AWS_CLI_ROOT}/bin/aws"
+RUN curl --fail "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "awscli-bundle.zip"
```

Comment:
Each `RUN` clause in a Dockerfile generates a separate intermediate layer. For related groups of commands, it is better to link the commands together with `&&` so they only end up generating a single layer.

Better would be:

```
RUN curl --fail -L "https://s3.amazonaws.com/aws-cli/awscli-bundle.zip" -o "awscli-bundle.zip" && \
    unzip awscli-bundle.zip && \
    ./awscli-bundle/install --install-dir "${AWS_CLI_ROOT}"
```

You should also use `-L` in case there are redirects.

Reply:
And from a software supply chain perspective, it would be better to pin to a particular version and check the downloaded archive's hash against a known value to ensure no tampering.

Reply:
Good to know! 

I'm going to stick with this here, I think, because this only gets built one time (by a Docker integration we have with GitHub) and when people very occasionally need to change the Docker image for CI (hasn't happened for 9 months). A separate line reads better imo. But, good to know for the future.

Reply:
To clarify, the intermediate layers survive the `docker build` and are pushed to the registry, and so there will be (probably) minimal impact on every use of this image having to merge the additional layers.

Reply:
> And from a software supply chain perspective, it would be better to pin to a particular version and check the downloaded archive's hash against a known value to ensure no tampering.

True. I'm gonna punt on that because this line is from their official docs and is identical to what we have in our script `install_aws_cli_for_ci.sh`. But, agreed that would be better.